File: p004.py
from euler import ispalindrom

def p004(a, b):
    numbers = [i * j for i in xrange(a, b + 1) for j in xrange(a, b + 1)]
    return max(filter(ispalindrom, numbers))

if __name__ == "__main__":
    from benchmark import benchmark
    benchmark(p004, 100, 999)

File: p004.clj
(use '[euler :only [run]])

(defn palindrome? [n]
  (let [s (str n)]
    (= s (clojure.string/reverse s))))

(defn products [digits]
  (for [a (range (Math/pow 10 (- digits 1)) (Math/pow 10 digits))
        b (range (Math/pow 10 (- digits 1)) (Math/pow 10 digits))]
    (int (* a b))))

(defn p004 [digits]
  (apply max (filter palindrome? (products digits))))

(run (p004 3))
